### ISSUE TITLE
Nightly struggles with Tuple{X}

### ISSUE THREAD

**Issue by SimonDanisch** [JuliaLang/julia#17247]
```Julia
abstract FixedArray{T, NDim, SIZE}
typealias FixedVector{CARDINALITY, T}        FixedArray{T, 1, Tuple{CARDINALITY,}}
abstract FixedVectorNoTuple{CARDINALITY, T} <: FixedVector{CARDINALITY, T}

immutable A <: FixedVectorNoTuple{3, Int} end
immutable B <: FixedVectorNoTuple{Val{3}, Int} end
immutable C <: FixedVector{3, Int} end
immutable D <: FixedVector{Val{3}, Int} end
immutable E{N, T} <: FixedVector{Val{N}, T} 
    x::NTuple{N, T}
end
immutable F{N, T} <: FixedVector{N, T} 
    x::NTuple{N, T}
end

julia> supertype(supertype(A))
FixedArray{Int64,1,Tuple{Any}}
julia> supertype(supertype(B))
FixedArray{Int64,1,Tuple{Any}}
julia> supertype(C)
FixedArray{Int64,1,Tuple{3}}
julia> supertype(D)
FixedArray{Int64,1,Tuple{Val{3}}}
julia> supertype(E{2, Int})
FixedArray{Int64,1,Tuple{Val{2}}}
julia> supertype(F{2, Int})
FixedArray{Int64,1,Tuple{Any}}

julia> versioninfo()
Julia Version 0.5.0-dev+5087
Commit eb05662* (2016-07-02 14:34 UTC)
Platform Info:
  System: NT (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i5-4200U CPU @ 1.60GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, haswell)
```

**Comment by andyferris** [comment]
Pure speculation: I wonder if this might have something to do with `Tuple{X}` where `X` is not a `DataType` or otherwise a type union or constructor. A long time ago I tried that trick regarding labelling indices for tensor notation, and I received some comments of surprise from core devs that you could even do that, and that it might not work/be allowed in the future. The problem is that there is no legal instance of `Tuple{2}`, etc, which is bizarre for a type (other than `Union{}`, which is special).

After that, I always used tuple *instances* like `(2,)` in type parameters, but I see that doesn't work well with your typealias for `FixedVector`. Is there a good workaround for this, like using `Vec{(N,), T}` and `Mat{(M,N),T}`? (as you observed, you can do `Tuple{Val{2}}` which has instance `(Val{2}(),)`, but that's getting uglier).

**Comment by SimonDanisch** [comment]
@andyferris, I think your observations are correct!
But still, this kind of undefined behaviour seems incorrect...

**Comment by SimonDanisch** [comment]
Someone seems to have fixed this... :)

**Comment by tkelman** [comment]
will pkgeval on fixedsizearrays catch it if it breaks again, or are there representative tests in base yet?

**Comment by c42f** [comment]
At the moment FixedSizeArrays will suffer from general complete breakage if this happens again, but the package design may change - there's some other options which don't abuse `Tuple` in quite the same way, such as SimonDanisch/FixedSizeArrays.jl#144 .

**Comment by tkelman** [comment]
mainly asking whether this should have the "needs-tests" label

**Comment by SimonDanisch** [comment]
Yeah, adding tests to base was the plan!

**Comment by c42f** [comment]
I ran a git bisect on this, and it picks out 089590b970 as the commit which fixed things, which makes approximately zero sense.  Here's the test script:

```bash
#!/bin/bash

make

./julia -e 'abstract FixedArray{T, NDim, SIZE}; typealias FixedVector{C,T} FixedArray{T,1,Tuple{C,}}; immutable F{N, T} <: FixedVector{N, T}; x::NTuple{N, T}; end; println(supertype(F{2,Int}).parameters[3])' | grep -v 'Any'
```

Run with
```
git bisect start 828f7ae efa23ad
git bisect run ./test.sh
```

**Comment by tkelman** [comment]
maybe add a `make clean` first?

**Comment by c42f** [comment]
I probably just got my bisect usage wrong - looks like @JeffBezanson fixed this one in 65e9bbefb4601

He included a test there, but it's rather different on the face of it.
